src/CodeGen_C.cpp

```diff
@@ -531,6 +531,29 @@ class ExternCallPrototypes : public IRGraphVisitor {
 };
 }
 
+void CodeGen_C::forward_declare_type(const Type &t) {
```

Comment:
I would consider calling this "declare_if_needed" or some such since a) doesn't declare all Types and b) it only does it the first time it is called on a type.